cadquery/FCAD_script_generator/molex/cq_models/conn_molex_502250.py

```diff
@@ -0,0 +1,188 @@
+#****************************************************************************
+
+__title__ = "model description for Molex 502250 series connectors"
+__author__ = "hackscribble"
+__Comment__ = 'model description for Molex 502250 series connectors using cadquery'
+
+___ver___ = "0.1 29/01/2018"
+
+class LICENCE_Info():
+    ############################################################################
+    STR_licAuthor = "Joel Holdsworth"
+    STR_licEmail = "joel@airwebreathe.org.uk"
+    STR_licOrgSys = ""
+    STR_licPreProc = ""
+
+    LIST_license = ["",]
+    ############################################################################
+
+
+import cadquery as cq
+from Helpers import show
+from collections import namedtuple
+import FreeCAD
+
+class series_params():
+    series = "502250"
+    manufacturer = 'Molex'
+    mpn_format_string = '502250-{pincount:02d}91'
+    orientation = 'H'
+    datasheet = 'http://www.molex.com/pdm_docs/sd/5022501791_sd.pdf'
+    pinrange = [17, 21, 23, 27, 33, 35, 39, 41, 51]
+    number_of_rows = 2
+    fp_name_format_string = '{man:s}_{mpn:s}_{num_rows:d}Rows-{pins:02d}Pins_P{pitch:.2f}mm_{orientation:s}'
+
+    pins_color_key = "metal grey pins"
+    body_color_key = "white body"
+    latch_color_key = "black body"
+    color_keys = [
+        body_color_key,
+        pins_color_key,
+        latch_color_key
+    ]
+    obj_suffixes = [
+        '__pins',
+        '__body',
+        '__latch'
+    ]
+
+    half_pitch = 0.3
+    pitch = half_pitch * 2
+
+    pin_row_spacing = 2.875
+    body_width = 3.53
+    body_height = 0.88
+    body_length_padding = 1.8
+    body_y_offset = 0.28 + pin_row_spacing/2
+    body_cut_width = body_width - 1.99
+
+
+def generate_pins(num_pins):
+    odd_pins = [
+        cq.Workplane("XY")
+            .moveTo(i * series_params.half_pitch - series_params.half_pitch * (num_pins - 1)/2,
+                -series_params.pin_row_spacing/2)
+            .box(0.12, 0.56, 0.21, centered=(True, True, False))
+        for i in range(0, num_pins, 2)]
+    even_pins = [
+        cq.Workplane("XY")
+            .moveTo(i * series_params.half_pitch - series_params.half_pitch * (num_pins - 1)/2,
+                series_params.pin_row_spacing/2)
+            .box(0.15, 0.40, 0.30, centered=(True, True, False))
+        for i in range(1, num_pins-1, 2)]
+    anchor_pins = [
+        cq.Workplane("XY")
+            .moveTo(d * ((num_pins-1) * series_params.half_pitch / 2 + 0.7),
+                    -series_params.pin_row_spacing/2 + 0.325)
+            .box(0.12, 0.55, 0.3, centered=(True, True, False))
+        for d in [-1, 1]]
+
+    pins = anchor_pins[0]
+    for p in [anchor_pins[1]] + odd_pins + even_pins:
+        pins = pins.union(p)
+    return pins.rotate((0, 0, 0), (0, 0, 1), -90)
+
+
+def generate_body(body_length):
+    slot_length = body_length - 0.6*2
+
+    body_outline = cq.Workplane("XY").polyline([
+        (0, 0.1),
```

Comment:
This line might be the trouble maker for the problem encountered by @easyw

Reply:
It seems in cadquerry 0.5.2 the polygone stuff always adds a point at 0,0.
the solution would be to first use moveTo.

If i change the lines to:
```
 body_outline = cq.Workplane("XY")\
        .moveTo(0, 0.1)\
        .polyline([
        #(0, 0.1),
        (body_length/2-0.745, 0.1),
```

it does work.